src/roi/Roi.ts

```diff
@@ -328,6 +329,34 @@ export class Roi {
     };
   }
 
+  get centroid() {
+    return this.#getComputed('centroid', () => {
+      const roiMap = this.getMap();
+      const data = roiMap.data;
+      let xCoords = [];
```

Comment:
We have here a special case. We know the length of the array at the beginning, it has to be the surface. Moreover in this case we know that it contains only integer. We should therefore use typed array Uint32. It will be in anyway much faster that using 'push'. https://jsbench.me/z8laj0dyuv